### ISSUE TITLE
Curious Blazor (Server) Behavior with Html Table Containing Input Elements

### ISSUE THREAD

**Issue by CincySteve** [dotnet/aspnetcore#50038]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

In a Blazor Server app I have a component that contains an html table with embedded input elements in each td cell, some showing previous user-entered data (via a value attribute and with CSS yellow highlighting) and the others showing default values (via a placeholder attribute) where the user hasn’t (yet) entered anything .

When I enter user data in a default (placeholder) cell that’s followed by 2 more default cells _in the same row_, the entered value ends up formatted in the 1st cell (as expected) and appears unformatted in the 2nd cell (unexpected). Also, if I enter the data using the tab key, the cursor ends up in the 3rd cell (unexpected), even though the onfocus events indicate the 2nd cell has focus. 

 I've stumbled on a workaround that's almost completely successful, though it suggests to me that the original problem is an issue in Blazor.  In particular, if I add if/else statements around the "placeholder" input markup to render identical input statements for even versus odd column numbers, the initial problem goes away, though focus still doesn't end up in the "2nd" cell.  Looking at the event logging shows a "focusin" to the "2nd" cell after I hit tab (expected), immediately followed by a "focusout" (unexpected), leaving no cell with focus. That "focusout" doesn't happen with the original code. This suggests to me that the original problem was due to the input in the 2nd cell becoming active with the data entered in the 1st cell as a preliminary value for the 2nd cell. With the workaround, the 2nd input doesn't remain active, thereby removing the preliminary input. 

P.S. I am using the “value” versus “placeholder” approach instead of using “value” for both to get around the situation where the user enters a value that matches the default, in which case no OnChange event is triggered that will to let me detect the user-entered value, which I need to detect.  I guess I could handle more fine-grained events to get around this, but yuck.

### Expected Behavior

I expect the entered value in the 1st cell with the cursor in the 2nd cell after hitting tab.

### Steps To Reproduce

I've created a GitHub [demo](https://github.com/CincySteve/BlazorTableInputDemo) that includes the workaround. You can toggle the workaround off and on by adding or removing the "true ||" in the if condition as noted in the code.  The demo is in the TableInputDemo page and the CSS is in site.css. To reproduce the problem, do the following after starting the app:

- Click on Table Input Demo in the Nav

- Enter 99 in the default cell that's showing $62 (1st cell) and hit tab.

- Notice that "$99" appears in the 1st cell (expected) and "99" (without the $) appears in the next/2nd cell (unexpected), which had shown $63 as a default placeholder. Also notice that the cursor ends up in the 3rd ($64) cell.

Look at the VS Output to see how the underlying data looks while you do this, including data from OnAfterRenderAsync events I've recorded.  

### .NET Version

.Net SDK 8.0.100-preview.7.23376.3 (x64) | ASP.NET Core 8.0.0 Preview 6 Build preview 6.23329.11 - Shared Framework (x64) | VS Community 2022 v17.7.0

**Comment by MackinnonBuck** [comment]
Thanks for reaching out. In order to determine if this is a bug in the framework, we'll need a more minimal repro. Could you please provide a shorter repro that demonstrates the bug (maybe ~10ish lines if possible)?

**Comment by CincySteve** [comment]
I've simplified the demo as much as I know how.  In the process, I learned more about the issue and have put 2 brief comments about that in the MainLayout markup, which is where I moved the demo code.  I hope this is simple enough.  I tried several further simplifications, but these did not experience the problem.  It seems to flow from my approach of using a data object (List of Lists) to generate the table and handle changes to it with subsequent re-rendering.  I am ready to do whatever else might be helpful to enable diagnosis of the problem.  Thanks for your help.

**Comment by surayya-MS** [comment]
Thank you for simplifying the repo. Unfortunately, there is still too much code and we cannot tell if there is a problem with the framework. We recommend to post your question on the stackoverflow. If you are able to track this down as a framework bug please let us know.

**Comment by CincySteve** [comment]
I'm discouraged to hear that because I posted this to SO 2 weeks ago because I know to start there (and I posted it in the MS Developer Community).  I have not had any response from either.  I'll try again with a different title.    This sure looks like a framework issue given the 2 variations I note in the MainLayout markup.  And in any case, if I am doing something wrong, it would seem there should be some indication of that instead of the app simply behaving in odd ways.  Anyway, I'll hope to get an SO response with another try.
